src/sentry/integrations/opsgenie/metrics.py

```diff
@@ -11,10 +11,56 @@ def record_event(event: OnCallInteractionType) -> OnCallInteractionEvent:
 # 40301 - To perform this action, use an API key from an API integration.
 OPSGENIE_HALT_ERRORS = (40301,)
 
+# Infrastructure-level error patterns that should be treated as halts rather than failures
+# These are typically temporary service availability issues outside our control
+OPSGENIE_INFRASTRUCTURE_ERROR_PATTERNS = (
+    "upstream connect error",
```

Comment:
Shouldn't it just be `upstream connect error or disconnect/reset before headers. reset reason: overflow` as a single error? I would also assume >= 500 error responses should always be halts too